app/login.tsx

```diff
@@ -5,13 +5,22 @@ import { router } from "expo-router";
 import { BACKGROUND_IMG, LOGO_ICON } from '../assets/images';
 import styles from "./loginStyles"
 import en from '@/translations/en.json';
+import { useState, useEffect } from 'react';
 
 export default function Login() {
-  const { signIn } = useSession();
+  const { signIn, session } = useSession();
   const handleLogin = () => {
-    signIn();
-    router.replace("/");
+    signIn(emailValue, passwordValue);
   };
+  const [emailValue, emailOnChange] = useState(en.login.emailPlaceholder);
+  const [passwordValue, passwordOnChange] = useState(en.login.passwordPlaceholder);
```

Comment:
This is perfectly fine, but it's important to point out react-hook-form, which will allow you to validate inputs, show error messages, and much more.